File: src/main/java/parser/ast/expressions/ArrayExpression.java
package parser.ast.expressions;

import stdlib.*;

import java.util.List;

public class ArrayExpression implements Expression{

    private final List<Expression> values;

    public ArrayExpression(List<Expression> values) {
        this.values = values;
    }

    @Override
    public Value eval() {
        final var size = values.size();
        final var array = new ArrayValue(size);
        for (var i = 0; i < size; i++) {
            array.set(i, values.get(i).eval());
        }

        return array;
    }

    @Override
    public String toString() {
        return String.format("<ArrayExpression><Items>%s</Items></ArrayExpression>", values.toString());
    }
}

File: src/main/java/parser/ast/expressions/ArrayExpression.kt
package parser.ast.expressions

import stdlib.ArrayValue
import stdlib.Value

class ArrayExpression(private val values: List<Expression>) : Expression {

    override fun eval(): Value {
        val size = values.size
        val array = ArrayValue(size)
        for (i in 0 until size) {
            array.set(i, values[i].eval())
        }

        return array
    }

    override fun toString(): String {
        return "<ArrayExpression><Items>$values</Items></ArrayExpression>"
    }
}
